### PR TITLE
Changed UI and front to support multiple filters

### PR DESCRIPTION
**Pull request by Noasan**

### CODE REVIEW

File: FrontEnd/src/main/java/resourceDisplay/IndexController.java [hunk 1]
```diff
@@ -46,29 +46,41 @@ public String index() {
...
+        List<String> locationList = assets.getFilterList(dbClient, "noasan", "location");
+        model.addAttribute("locationList", locationList);
+        List<String> statusList = assets.getFilterList(dbClient, "noasan", "status");
+        model.addAttribute("statusList", statusList);
+        List<String> kindList = assets.getFilterList(dbClient, "noasan", "kind");
+        model.addAttribute("kindList", kindList);
+        String status = filterObject.getStatus();
+        String location = filterObject.getLocation();
+        String kind = filterObject.getKind();
+        if(status != null && location != null && kind != null) {
```
Comment (nirfi) [inline FrontEnd/src/main/java/resourceDisplay/IndexController.java]: whitespace between `if` and `(`

File: FrontEnd/src/main/java/resourceDisplay/IndexController.java [hunk 2]
```diff
@@ -46,29 +46,41 @@ public String index() {
...
         spanner = options.getService();
         db = DatabaseId.of(SPANNER_PROJECT_ID, SPANNER_INSTANCE_ID, SPANNER_DATABASE_ID);
         dbClient = spanner.getDatabaseClient(db);
         AssetsRepository assets = new AssetsRepository();
-        //TODO use workspaceID from user/workspace/project
-        ResultListObject resultListObject = assets.getAllAssets(dbClient);
-        model.addAttribute("displayNames", resultListObject.columnDisplays);
-        model.addAttribute("allAssets", resultListObject.columnResults);
+        model.addAttribute("filterObject", filterObject);
+        //TODO use real workspaceID from user/workspace/project table this is temporary (using "noasan")
```
Comment (nirfi) [inline FrontEnd/src/main/java/resourceDisplay/IndexController.java:66]: any idea when are we planning to move ahead with this TODO?

File: FrontEnd/src/main/java/resourceDisplay/IndexController.java [hunk 3]
```diff
@@ -46,29 +46,41 @@ public String index() {
...
-     * @param model     - Used to show table to user
-     * @return the allassets template
+     * @param principal    - Used to check authentication
+     * @param model        - Used to show table to user
+     * @param filterObject - Used to get filters from the user
+     * @return the bystatus template
      */
     @GetMapping("/allassets")
-    public String getAll(@AuthenticationPrincipal OAuth2User principal, Model model) {
+    public String test(@AuthenticationPrincipal OAuth2User principal, Model model,
```
Comment (nirfi) [inline FrontEnd/src/main/java/resourceDisplay/IndexController.java]: change method name

File: FrontEnd/src/main/java/resourceDisplay/AssetsRepository.java [hunk 4]
```diff
@@ -237,11 +237,18 @@ public TableQueryObject createTableQueryObject(DatabaseClient dbClient, String f
...
         String statementString;
-        if (filterType != null) {
-            statementString = String.format("SELECT %s FROM Main_Assets WHERE %s = @%s"
-                    , columns, filterType, filterType);
+        if (filters.isEmpty()) {
+            statementString = String.format("SELECT %s FROM Main_Assets ORDER By %s", columns,
+                    columnNames.get(0));
         } else {
-            statementString = String.format("SELECT %s FROM Main_Assets", columns);
+            List<String> forWhere = new ArrayList<>();
```
Comment (nirfi) [inline FrontEnd/src/main/java/resourceDisplay/AssetsRepository.java]: a name like `queryFilters` would explain better what you're collecting here

**Review by nirfi — CHANGES_REQUESTED** [review]

File: FrontEnd/src/main/java/resourceDisplay/IndexController.java (hunk 1 above)
Comment (Noasan) [reply FrontEnd/src/main/java/resourceDisplay/IndexController.java]: done:)

File: FrontEnd/src/main/java/resourceDisplay/IndexController.java (hunk 2 above)
Comment (Noasan) [reply FrontEnd/src/main/java/resourceDisplay/IndexController.java:66]: Will work on this after we finish creating the service account table (working on it now).

File: FrontEnd/src/main/java/resourceDisplay/IndexController.java (hunk 3 above)
Comment (Noasan) [reply FrontEnd/src/main/java/resourceDisplay/IndexController.java]: done

File: FrontEnd/src/main/java/resourceDisplay/AssetsRepository.java (hunk 4 above)
Comment (Noasan) [reply FrontEnd/src/main/java/resourceDisplay/AssetsRepository.java]: done

**Review by nirfi — APPROVED** [review]

**Pull request merged**